src/compiler.jl

```diff
@@ -3185,19 +3192,22 @@ end
 else
             jctx = ctxToThreadSafe[jctx]
 end
-            cmod, adjointnm, augfwdnm, _, TapeType = _thunk(ejob, jctx, #=postopt=#false)
+            cmod, adjointnm, augfwdnm, TapeType = context!(jctx) do
```

Comment:
I don't think we should need this since we should preumably already be in a context?